generic3g/actions/VerticalRegridActionNew.F90

```diff
@@ -0,0 +1,75 @@
+#include "MAPL_Generic.h"
+
+module mapl3g_VerticalRegridActionNew
+
+   use mapl_ErrorHandling
+   use mapl3g_ExtensionAction
+   use mapl3g_VerticalRegridMethod, only: VerticalRegridMethod_Flag
+   use mapl3g_CSR_SparseMatrix
+   use esmf
+   use, intrinsic :: iso_fortran_env, only: REAL32
+
+   implicit none
+   private
+
+   public :: VerticalRegridAction
+
+   type, extends(ExtensionAction) :: VerticalRegridAction
+      real(REAL32), allocatable :: src_vertical_coord(:)
+      real(REAL32), allocatable :: dst_vertical_coord(:)
```

Comment:
Note for next pass:   We need to support REAL32 and REAL64 in this layer.  (Or alternatively by having 2 different subclasses, but I think that will be more difficult to maintain.)